### PR TITLE
refactor(ProductiveCard): add Typescript types to ProductiveCard

### LINKED ISSUE #4505: Add Typescript types to ProductiveCard
As we add Typescript support, want to call out a few helpful tips from the Carbon TypeScript guide that is largely relevant to our library as well.

### Changing the extension via git

When changing the file extension — **do not simply rename the file**. **Instead, change the extension via `git`.**

For example,

```
git mv packages/ibm-products/src/components/ComponentName/ComponentName.js packages/ibm-products/src/components/ComponentName/ComponentName.tsx
```

### Do not add types to internal components or functions
- Leave internals as `.js`
- Add a [jsdoc type annotation](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#providing-type-hints-in-js-via-jsdoc) of `/** @type any */`
- [Enable errors](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check) by adding `// @ts-check` to the first line in the file.

### Testing changes
A few different options for testing changes —

- At the bottom of a file, write a dummy component that uses the component that you’re converting to ensure that you can still pass all appropriate props.
- Take a storybook example for a component, copy and paste it into the bottom of the `.tsx` file and validate whether or not it accepts the props as you’ve defined them.

```[tasklist]
### Tasks
- [ ] Change extension to `.tsx` **via git** (**do not simply rename**, see above)
- [ ] Copy the PropTypes definitions above the component definition
- [ ] Retool the PropTypes to be a TypeScript interface
- [ ] Fix errors as they appear
- [ ] Do NOT add types to internal components or functions that are not exported as part of the public API
- [ ] Test your changes (see above)
```

### Resources
Carbon’s guide on [adding component types](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md)

### PR DESCRIPTION
**Pull request by amal-k-joy**
Closes #4505 

Add Typescript types to ProductiveCard

#### What did you change?
file extension
TS interface

#### How did you test and verify your work?
storybook build and type checks

### CODE REVIEW

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx [hunk 1]
```diff
@@ -5,23 +5,137 @@
...
-const defaults = {
-  actionsPlacement: 'top',
+type ActionIcon = {
+  id: string;
+  icon: CarbonIconType;
+  onKeyDown?(): void;
+  onClick?(): void;
+  iconDescription: string;
+  href?: string;
 };
```
Comment (anamikaanu96) [inline packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx:26]: Please make id, icon and iconDescription of ActionIcon to optional.

Make sure the import statement of CarbonIconType is correct. Now it is coming from '@carbon/react/icons'

**Review by anamikaanu96 — CHANGES_REQUESTED** [review]

**Review by anamikaanu96 — CHANGES_REQUESTED** [review]
Can you please add children in ProductiveCardProps interface since it is mandatory. Now its considering children as optional.

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx [hunk 2]
```diff
@@ -5,23 +5,137 @@
...
+  // children: PropTypes.node,
+  /**
+   * Optional user provided class
+   */
+  className?: string;
+
+  /**
+   * Designates which zones of the card are clickable. Refer to design documentation for implementation guidelines
+   */
+  clickZone?: 'one' | 'two' | 'three';
```
Comment (makafsal) [inline packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx:55]: Better make this as a separate type, may be can reuse in future.

**Review by makafsal — CHANGES_REQUESTED** [review]

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx (hunk 1 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx:31]: modified as advised

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx [hunk 3]
```diff
@@ -5,23 +5,137 @@
...
+interface ProductiveCardProps extends PropsWithChildren {
+  /**
+   * Icons that are displayed on card. Refer to design documentation for implementation guidelines. Note- href will supersede onClick
+   */
+  actionIcons?: ActionIcon[];
+
+  /**
+   * Determines if the action icons are on the top or bottom of the card
+   */
+  actionsPlacement?: 'top' | 'bottom';
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx]: modified as advised

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx [hunk 4]
```diff
@@ -5,23 +5,137 @@
...
+   * Determines title size
+   */
+  titleSize?: 'default' | 'large';
+}
 
 export let ProductiveCard = forwardRef(
-  ({ actionsPlacement = defaults.actionsPlacement, ...rest }, ref) => {
+  (
+    { actionsPlacement = 'top', ...rest }: ProductiveCardProps,
+    ref: React.Ref<HTMLDivElement>
```
[the remark this answers is not in the document]
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx]: modified as advised

File: packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx (hunk 2 above)
Comment (amal-k-joy) [reply packages/ibm-products/src/components/ProductiveCard/ProductiveCard.tsx]: modified as advised

**Review by anamikaanu96 — APPROVED** [review]

**Review by makafsal — APPROVED** [review]

**Pull request merged**